src/Validator/Constraints/EntityValidator.php

```diff
@@ -89,170 +72,18 @@ public function validate($entity, Constraint $constraint): void
-                break;
-            case 'longitude':
-                $constraints[] = new Longitude();
-                $constraints[] = new FloatScale($fieldMapping['scale'] ?? LongitudeType::DEFAULT_SCALE);
-                break;
-            case 'money':
-                $constraints[] = new Money();
-                $constraints[] = new FloatScale($fieldMapping['scale'] ?? MoneyType::DEFAULT_SCALE);
-                break;
-            case 'percent':
-                $constraints[] = new Type(\AssoConnect\PHPPercent\Percent::class);
-                $constraints[] = new Percent();
-                break;
-            case 'phone':
-                $constraints[] = new Phone();
-                break;
-            case 'phonelandline':
-                $constraints[] = new PhoneLandline();
-                break;
-            case 'phonemobile':
-                $constraints[] = new PhoneMobile();
-                break;
-            case 'postal':
-                break;
-            case 'simple_array':
-                $constraints[] = new Type('array');
-                break;
-            case 'smallint':
-                $constraints[] = new Type('integer');
-                if (isset($fieldMapping['options']['unsigned']) && true === $fieldMapping['options']['unsigned']) {
-                    $constraints[] = new GreaterThan(0);
-                    $constraints[] = new LessThanOrEqual(pow(2, 16) - 1);
-                } else {
-                    $constraints[] = new GreaterThanOrEqual(- pow(2, 15));
-                    $constraints[] = new LessThanOrEqual(pow(2, 15) - 1);
-                }
-                break;
-            case 'string':
-                $length = $fieldMapping['length'] ?? 255;
-                $constraints[] = new Length(['max' => $length]);
-                if (isset($fieldMapping['nullable']) && true === $fieldMapping['nullable']) {
-                    $constraints[] = new NotBlank(['allowNull' => true]);
-                }
-                break;
-            case 'text':
-                $length = $fieldMapping['length'] ?? 65535;
-                $constraints[] = new Length(['max' => $length, 'charset' => '8bit']);
-                break;
-            case 'timezone':
-                $constraints[] = new Timezone();
-                break;
-            case 'ulid':
-                $constraints[] = new Ulid();
-                break;
-            case 'uuid':
-            case 'uuid_binary_ordered_time':
-                $constraints[] = new Uuid();
-                break;
-            default:
-                throw new \DomainException('Unsupported field type: ' . $fieldMapping['type']);
+        foreach ($this->fieldConstraintsSetFactories as $fieldConstraintsSetFactory) {
+            if ($fieldConstraintsSetFactory->supports($fieldMapping['type'])) {
+                return $fieldConstraintsSetFactory->getConstraints($fieldMapping);
```

Comment:
We take the first set of constraints found and with this behavior, we can override the constraints defined. Is it better to override or merge constraints found?